jbmc/src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1655,6 +1658,32 @@ void java_bytecode_parsert::rinner_classes_attribute(
   }
 }
 
+/// Corresponds to the element_value structure
+/// Described in Java 8 specification 4.7.5
+/// https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.7.5
+/// Parses the Exceptions attribute for the current method,
+/// which contains a table of the exceptions the method throws.
+void java_bytecode_parsert::rexceptions_attribute(methodt &method, const u4 &attribute_length)
+{
+  std::string name = method.name.c_str();
+  u2 number_of_exceptions = read_u2();
+  method.throws_exception_table.resize(number_of_exceptions);
+
+  const auto pool_entry_lambda = [this](u2 index) -> pool_entryt & {
+    return pool_entry(index);
+  };
+
+  if (number_of_exceptions == 0)
+    return;
+  for(size_t i = 0; i < number_of_exceptions; i++)
+  {
+    u2 exception_index_table=read_u2();
+    pool_entryt pool_entry_temp=pool_entry_lambda(exception_index_table);
+    const irep_idt temp = constant(exception_index_table).type().get(ID_C_base_name);
+    method.throws_exception_table.push_back(temp);
+  }
+}
+
```

Comment:
There are so many unused variables, the body of this function could be rewritten as:

```c++
const u2 number_of_exceptions = read_u2(); 
for(size_t i = 0; i < number_of_exceptions; i++)
  method.throws_exception_table.push_back(
    constant(read_u2()).type().get(ID_C_base_name));
```

I'm not sure about `.resize()` - do you want to shrink the vector, fill it with empty elements or [`.reserve()`](http://www.cplusplus.com/reference/vector/vector/reserve/) elements to make push_back more efficient (in which case it's not worth it here I think)?

Rather than passing the method as an argument, you could return a vetor of constants. Right now it's not clear whether you're appending exceptions to an existing vector or creating new one from scratch. This forces the reader to look elsewhere whether this vector contains anything or not.

Also, this function doesn't seem to modify the state, so could be marked as `const`, but for that you'd have to mark `read_u2` and `read_bytes` as const (I tried that, and it compiles).

Reply:
Actually, not everything is const. There were some methods further down the chain that were required to be const and couldn't be.